app/config.go

```diff
@@ -27,3 +49,31 @@ var evmosActivators = map[string]func(*vm.JumpTable){
 	"evmos_1": eips.Enable0001,
 	"evmos_2": eips.Enable0002,
 }
+
+// setBaseDenomWithChainID registers the display denom and base denom and sets the
+// base denom for the chain. The function registers different values based on
+// the chainID to allow different configurations in mainnet and testnet.
+func setBaseDenomWithChainID(chainID string) {
+	// only set to aevmos on testnet
+	if utils.IsTestnet(chainID) {
+		if err := sdk.RegisterDenom(types.DisplayDenomTestnet, math.LegacyOneDec()); err != nil {
+			panic(err)
+		}
+		if err := sdk.RegisterDenom(types.BaseDenomTestnet, math.LegacyNewDecWithPrec(1, types.BaseDenomUnit)); err != nil {
+			panic(err)
+		}
+		if err := sdk.SetBaseDenom(types.BaseDenomTestnet); err != nil {
+			panic("can't set base denom")
+		}
+		return
```

Comment:
Can we have this in a more structured way? like a map of configs by chain-id? something silimar to [geth](https://github.com/ethereum/go-ethereum/blob/deb5c087c41e5d7692fac2cff4c5aa547a108bc1/params/config.go#L40)
So we are not restricted with the utils.IsTesnet() when setting the config. Will be just checking if the chain id is within the available configs and set the corresponding config.